types/credentials.go

```diff
@@ -1,11 +1,17 @@
 package types
 
 import (
+	"log"
 	"os"
 
 	"github.com/openfaas/faas-provider/auth"
 )
 
+// GetCredentials returns a pointer to basic auth credentials using environment
+// variable lookup for runtime secrets retrieved via "secret_mount_path"
+// environment variable.
+//
+// It panics if credentials cannot be retrieved.
 func GetCredentials() *auth.BasicAuthCredentials {
```

Comment:
Personally I would like to see this error bubbled up in idiomatic go style and then handled at the top level, what do you think @embano1 ?